packages/node-sdk/src/client.ts

```diff
@@ -0,0 +1,607 @@
+      event,
+      companyId: this._company?.companyId,
+      userId: this._user?.userId,
+      attributes: opts?.attributes,
+      context: opts?.meta,
+    });
+  }
+
+  /**
+   * Initializes the client by caching the feature flag definitions.
+   *
+   * @returns void
+   *
+   * @remarks
+   * Call this method before calling `getFlags` to ensure the feature flag definitions are cached.
+   **/
+  public async initialize() {
+    await this.getFeatureFlagDefinitionCache().refresh();
+  }
+
+  /**
+   * Gets the evaluated feature flags for the current context which includes the user, company, and custom context.
+   *
+   * @typeparam TFlagKey - The type of the feature flag keys, `string` by default.
+   *
+   * @returns The evaluated feature flags.
+   * @remarks
+   * Call `initialize` before calling this method to ensure the feature flag definitions are cached, empty flags will be returned otherwise.
+   **/
+  public getFlags<TFlagKey extends string = string>(): Readonly<
+    Record<TFlagKey, Flag>
+  > {
+    const mergedContext = {
+      user: this._user && {
+        id: this._user.userId,
+        ...this._user.attrs,
+      },
+      company: this._company && {
+        id: this._company.companyId,
+        ...this._company.attrs,
+      },
+      ...this._otherContext,
+    };
+
+    const flagDefinitions = this.getFeatureFlagDefinitionCache().get();
+    let evaluatedFlags: Flags<TFlagKey> =
+      (this._shared.fallbackFlags as Flags<TFlagKey>) || {};
+
+    if (flagDefinitions) {
+      const keyToVersionMap = new Map<string, number>(
+        flagDefinitions.flags.map((f) => [f.key, f.version]),
+      );
+
+      const evaluated = flagDefinitions.flags.map((flag) =>
+        evaluateFlag({ context: mergedContext, flag }),
+      );
+
+      evaluated.forEach(async (res) => {
+        await this.sendFeatureFlagEvent({
+          action: "evaluate",
+          flagKey: res.flag.key,
+          flagVersion: keyToVersionMap.get(res.flag.key),
+          evalResult: res.value,
+          evalContext: res.context,
+          evalRuleResults: res.ruleEvaluationResults,
+          evalMissingFields: res.missingContextFields,
+        });
+      });
+
+      evaluatedFlags = evaluated
+        .filter((e) => e.value)
+        .reduce(
+          (acc, res) => {
+            acc[res.flag.key as TFlagKey] = {
+              key: res.flag.key,
+              value: res.value,
+              version: keyToVersionMap.get(res.flag.key),
+            };
+            return acc;
+          },
+          {} as Record<TFlagKey, Flag>,
+        );
+
+      this._shared.logger?.debug("evaluated flags", evaluatedFlags);
+    } else {
+      this._shared.logger?.warn(
+        "failed to use feature flag definitions, there are none cached yet. using fallback flags.",
+      );
+    }
+
+    return readNotifyProxy(
+      evaluatedFlags,
+      rateLimited(FLAG_EVENTS_PER_MIN, async (_: string, res: Flag) => {
```

Comment:
This should be using the same cache key calculation as in the tracking-sdk. E.g. send a check event per flag.key, context, version ... per minute